src/Tribe/Validate.php

```diff
@@ -252,7 +252,14 @@ public function int() {
 		 * @return stdClass validation result object
 		 */
 		public function slug() {
-			if ( preg_match( '/^[a-zA-Z0-9-_]+$/', $this->value ) ) {
+			$maybe_valid_value = esc_url_raw( $this->value );
+
+			// While esc_url_raw is handy and does the work of validating for us, it returns the
+			// checked string with a prepended URL protocol; let's remove that.
+			$maybe_valid_value = wp_kses_bad_protocol_once( $maybe_valid_value, array() );
+			$maybe_valid_value = str_replace( '//', '', $maybe_valid_value );
+
+			if ( $maybe_valid_value == $this->value ) {
```

Comment:
Just a thought: can we skip the juggling by just doing a `stripos`?

Reply:
@Camwyn

we _could_, but only by explicitly naming the protocols 

my thinking is that, since this validation method could in theory be used anywhere, in lots of diff contexts, we want to make sure there's absolutely no protocol there, and i wanted to cover anything beyond `http://` and `https://` so we are *certainly* dealing with a slug only

Reply:
Gotcha.

Reply:
@Camwyn and i are discussing this choice some more over Slack DM. may change approach here.